src/DynamoCoreWpf/ViewModels/Preview/WatchViewModel.cs

```diff
@@ -274,5 +275,40 @@ public void CountLevels()
         {
             Levels = maxListLevel > 0 ? Enumerable.Range(1, maxListLevel).Reverse().Select(x => x).ToList() : Enumerable.Empty<int>();
         }
+
+        /// <summary>
+        /// Get the NodeLabel for this WatchViewModel and any children.
+        /// Each level of children will be indented by 2 spaces.
+        /// </summary>
+        /// <param name="depth">The number of levels of indentation.</param>
+        /// <param name="includeKey">If true the list or dictionary key will be included in the string.</param>
+        /// <returns></returns>
+        public string GetNodeLabelTree(int depth = 0, bool includeKey = false)
```

Comment:
@mmisol do you think this should be public or is internal fine?

Reply:
It's accessed from another project, so it should be either `public` or `internal` using `InternalsVisibleTo`. I think I have expressed it before but I don't think any `public` member is part of our API. We should document exactly which parts are.